src/main/java/htsjdk/samtools/reference/FastaSequenceFile.java

```diff
@@ -89,8 +105,16 @@ public ReferenceSequence nextSequence() {
     public void reset() {
         this.sequenceIndex = -1;
         this.in.close();
-        this.in = new FastLineReader(IOUtil.openFileForReading(getPath()));
-
+        if (getPath() != null) {
+            this.in = new FastLineReader(IOUtil.openFileForReading(getPath()));
```

Comment:
This if clause can be removed if the `Path` in open on construction as a `SeekableStream`